### DISCUSSION TITLE
Is `crypto_secretbox_xchacha20poly1305_easy` safe to be used?

### DISCUSSION THREAD

**Discussion by chirieac** [jedisct1/libsodium#1101] [Q&A] [answered]
Hi,

Why `crypto_secretbox_xchacha20poly1305_easy` function is not documented anywhere, isn't recommended to be used?

**libsodium.js** doesn't expose the function, and neither PHP (it's not in its [documentation](https://www.php.net/manual/en/book.sodium.php) at least), which makes me wonder why.

If we want **XChaCha20** instead of **XSalsa20** we should use `crypto_aead_xchacha20poly1305_ietf_encrypt` instead, even if we do not care about _additional data_?

**Comment by jedisct1** [comment, marked as answer]
`secretbox_xchacha20poly1305` is here for consistency, but is not used by anything and is likely to eventually be removed.

If you already have a shared secret, the high-level `secretbox` is compatible with all reimplementations of NaCl.

There's no difference in security between full rounds salsa20 and chacha20, so there are very little reasons to use `secretbox_xchacha20poly1305`. Its format is also incompatible with `xchacha20poly1305` as implemented as an AEAD everywhere else.

So, if you explicitly need to use XChaCha20, yes, you should use `aead_xchacha20poly1305` instead. The additional data can be empty.

**Reply by chirieac** [reply]
Understood, thank you!

If there's no difference in security between **XSalsa20** and **XChaCha20**, then maybe I should use the high-level `secretbox` and move on.

I've been trying to understand all these a little better before choosing. Based on my research, without having much experience with cryptographic algorithms, I got the assumption that **XChaCha20** is better being a successor of **XSalsa20**. Though, I haven't found any clear reason (at least that I understand) why would someone use _XChaCha20_ over _XSalsa20_, why it is better.

People seems to talk about _XChaCha20_ as alternative to **AES**, so I thought it might be better, more future-proof.

**Reply by jedisct1** [reply]
Salsa20 is what the author of both still recommends over ChaCha20. Both are very similar, and if a major attack affects Salsa, it will likely also affect ChaCha and many others. But there are no signs of any forthcoming practical attack against against any of these ciphers. AES is unlikely to ever be broken either.

**Reply by chirieac** [reply]
Thank you for your further explanations!

But what you said raises another question in my head: if both are very similar and the author still recommends Salsa20, why there's only ChaCha20 implementation for AEAD construction? Why it was chosen over Salsa20?
